ConditionalTranslation.md

```diff
@@ -0,0 +1,297 @@
+
+A *translate-time attribute* can appear in all places where the WGSL grammar allows attributes, except if removal of the decorated syntax node would lead to syntactically incorrect code.
+The grammar is extended to allow attributes in several locations previously not allowed by the WGSL grammar. These extensions are specified in section [updated grammar](#updated-grammar).
+
+### Summary
+
+1. The grammar is extended to allow *translate-time attributes* before the following syntax nodes:
+   * const value declarations
+   * variable declarations
+   * directives
+   * switch clauses
+   * else if clauses and else clauses
+   * all statements that did not allow attributes:
+     * assignment statements
+     * increment and decrement statements
+     * break statements
+     * break-if statements
+     * continue statements
+     * continuing statements
+     * return statements
+     * discard statements
+     * function call statements
+     * const assertion statements
+
+2. A *translate-time attribute* CAN decorate the following syntax nodes:
+   * structure declarations
+   * structure members
+   * function declarations
+   * function formal parameter declarations
+   * variable declarations and value declarations
+   * all statements, except those disallowed
+   * directives
+
+3. A *translate-time attribute* CANNOT decorate the following syntax nodes, even if the WGSL grammar allows attributes before these syntax nodes:
+   * function return types
+   * the body (part surrounded by curly braces) of:
+     * function declarations
+     * switch statements
+     * switch clauses
+     * loop statements
+     * for statements
+     * while statements
+     * if/else statements
+
+## `@if` attribute
+
+A new *translate-time attribute* is introduced.
+* An `@if` attribute takes a single parameter. It marks the decorated node for removal if the parameter evaluates to `false`.
+
+> NOTE: See the [possible future extensions](#possible-future-extensions) for the attributes `@elif` and `@else`.
+> They may be introduced in the specification in a future version if deemed useful.
+
+*Example*
+
+```rs
+@if(feature_1 && (!feature_2 || feature_3))
+fn f() { ... }
+@if(!feature_1)                               // corresponds to @elif(!feature_1)
+fn f() { ... }
+@if(feature_1 && !(!feature_2 || feature_3))  // corresponds to @else
+fn f() { ... }
+```
+
+## Execution of the conditional translation phase
+
+The conditional translation phase *should* be the first phase to run in the full WESL-to-WGSL translation pipeline.
+
+> NOTE: The conditional translation was designed to be incremental. In case some features can only be resolved at runtime, the *WESL translator* can be invoked in multiple passes:
+> A first pass is invoked with features determined at compile-time and returns a partially translated WESL source.
+> A second pass is invoked with features determined at run-runtime and returns a fully translated WGSL source.
```

Comment:
Damn, I wish we had better names for that. Right now I can't think of a significantly better option though.